### PR TITLE
[e2e] add codeowners for new e2e tests

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Add codeowners for new e2e tests code introduced by https://github.com/DataDog/datadog-agent/pull/13643, assigning it to @DataDog/agent-e2e-testing 

### Motivation

Prevent @DataDog/agent-platform from being asked on our code's PRs, move ownership to proper owners

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (KSerrania) [inline .github/CODEOWNERS]:
```suggestion
/test/new-e2e/                          @DataDog/agent-e2e-testing
```

for consistency with the rest of the file

**Review by KSerrania — APPROVED** [review]
LGTM. It looks like the codeowners linter is complaining: 
image

I'll give write access for that team to the repository to make the error go away.

**Comment by KSerrania** [comment]
Done ✅

**Comment by pducolin** [comment]
No changes in my forced push, rebased on [vboulineau/e2e-tests-demo](https://github.com/DataDog/datadog-agent/tree/vboulineau/e2e-tests-demo)

**Comment by pducolin** [comment]
Again forced push a rebase on main, after `vboulineau/e2e-tests-demo` merge

**Pull request merged**
